onnx/optimizer/pass.h

```diff
@@ -0,0 +1,179 @@
+// ATTENTION: The code in this file is highly EXPERIMENTAL.
+// Adventurous users should note that the APIs will probably change.
+
+#pragma once
+
+#include <string>
+#include "onnx/common/ir.h"
+#include "onnx/onnx_pb.h"
+
+namespace ONNX_NAMESPACE {
+namespace optimization {
+
+// Base struct representing result of a pass.
+struct PostPassAnalysis {};
+
+// Enum that represents the type of optimization it is.
+enum PassType {
+  // Class of optimizations that fuses operations.
+  Fuse = 0,
+  // Class of optimizations that remove useless operations.
+  Nop = 1,
+  // Class of optimizations that includes some form of seperation.
+  Seperate = 2,
+  // Immutable pass, also sometimes referred to as an analysis pass.
+  Immutable = 3,
+  // Unknown type of pass.
+  Unk = 4
```

Comment:
To me, unknown sounds better, shall we call it others?

Reply:
I think Other is better. A user will not write a pass that is of unknown type but rather other.